apps/ops/ws.py

```diff
@@ -51,13 +71,17 @@ def get_task(self, task_id):
 
     @sync_to_async
     def get_current_user_role_ids(self, user):
-        roles = user.system_roles.all() | user.org_roles.all()
+        with tmp_to_org(self.org):
+            org_roles = user.org_roles.all()
+        system_roles = user.system_roles.all()
+        roles = system_roles | org_roles
         user_role_ids = set(map(str, roles.values_list('id', flat=True)))
         return user_role_ids
 
     @sync_to_async
     def has_perms(self, user, perms):
-        return user.has_perms(perms)
+        with tmp_to_org(self.org):
+            return user.has_perms(perms)
 
     async def receive_json(self, content, **kwargs):
         task_id = content.get('task')
```

Comment:
There doesn't appear to be any significant code differences that call out specific bugs or inefficiencies. The main changes introduced in this file seem to revolve around the use of `SimpleCookie` for managing cookies, which can affect browser compatibility. For more comprehensive reviews, you might want to analyze the logs and database queries involved rather than just looking at the code structures themselves.

However, since there were no issues reported before 2021-09-01 (which is now), it's likely some other area would have been impacted by an earlier change that may warrant scrutiny:

* If performance was affected due to the complexity in parsing the cookie string.
* There may be inconsistencies between different environments or versions leading to unexpected behaviors.

As a result, my current focus will be on ensuring consistent testing across platforms and identifying areas needing improvement. I recommend reviewing the logs from older deployments as well using Python's built-in logging capabilities (`logging.info()` etc.) instead of importing modules to help isolate any patterns affecting log output.

If necessary, additional debugging steps like inspecting variables during runtime could also provide useful insights into where optimizations and updates are needed.